File: instaCults/Models/Cult.cs
namespace instaCults.Models;

public class Cult
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Tags { get; set; }
    public string Description { get; set; }
    public int Popularity { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string LeaderId { get; set; }
    public Profile Leader { get; set; }

}

File: instaCults.client/src/models/Cult.js
import { Profile } from "./Account.js"

export class Cult {
    constructor(data) {
        this.id = data.id
        this.name = data.name
        this.tags = data.tags
        this.description = data.description
        this.popularity = data.popularity
        this.coverImg = data.coverImg
        this.createdAt = new Date(data.createdAt)
        this.updatedAt = new Date(data.updatedAt)
        this.leaderId = data.leaderId
        this.leader = new Profile(data.leader)
    }
}

